### PR TITLE
feat(ext/net): add `NS` DNS record support

### LINKED ISSUE #14355: Deno panicked with Deno.resolveDns trying to resolve NameServer(NS)
Command to reproduce the problem:
await Deno.resolveDns("deno.land","NS")

=========================Message Error===================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: linux x86_64
Version: 1.19.3
Args: ["deno"]

thread 'main' panicked at 'not yet implemented', /home/runner/work/deno/deno/ext/net/ops.rs:736:12
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

### PR DESCRIPTION
**Pull request by PatrickChoDev**
Adding `NS` DNS record using `Name` struct. For resolving name server record to delegates a DNS zone to use the given with [RFC 1035](https://datatracker.ietf.org/doc/html/rfc1035#page-12).
Closes #14355

### CODE REVIEW

**Comment by bartlomieju** [comment]
Looks good, could you add a test case to `test_resolve_dns` integration test?

File: cli/tests/integration/mod.rs [hunk 1]
```diff
@@ -1003,6 +1012,7 @@ async fn test_resolve_dns() {
       .unwrap();
     let err = String::from_utf8_lossy(&output.stderr);
     let out = String::from_utf8_lossy(&output.stdout);
+    println!("{}", err);
```
Comment (bartlomieju) [inline cli/tests/integration/mod.rs:1015]: Remove

**Review by bartlomieju — COMMENTED** [review]
Looks good, let's see if CI passes

File: cli/dts/lib.deno.ns.d.ts [hunk 2]
```diff
@@ -2953,6 +2953,7 @@ declare namespace Deno {
     | "ANAME"
     | "CNAME"
     | "MX"
+    | "NS"
```
Comment (bartlomieju) [inline cli/dts/lib.deno.ns.d.ts:2956]: A new overload for `resolveDns` should be added to this file as well

**Review by bartlomieju — CHANGES_REQUESTED** [review]

File: cli/dts/lib.deno.ns.d.ts (hunk 2 above)
Comment (bartlomieju) [reply cli/dts/lib.deno.ns.d.ts:2956]: Actually, a new overload is probably not needed, but "NS" variant should be added to one of existing overloads.

Comment (PatrickChoDev) [reply cli/dts/lib.deno.ns.d.ts:2956]: resolved in 6ccb2ce03ca1ffe3d298dfda028124525d7848a4

File: cli/tests/integration/mod.rs (hunk 1 above)
Comment (PatrickChoDev) [reply cli/tests/integration/mod.rs]: removed in 6ccb2ce03ca1ffe3d298dfda028124525d7848a4

**Review by bartlomieju — APPROVED** [review]
LGTM, nice work @PatrickChoDev!

**Pull request merged**